rom/dev/tools/test-fmc/src/main.rs

```diff
@@ -178,3 +183,74 @@ fn copy_tbs(tbs: &mut [u8], ldevid_tbs: bool) {
     };
     tbs.copy_from_slice(src);
 }
+
+fn get_pcr_entry(entry_index: usize) -> PcrLogEntry {
+    // Copy the pcr log entry from DCCM
+    let mut pcr_entry: [u8; core::mem::size_of::<PcrLogEntry>()] =
+        [0u8; core::mem::size_of::<PcrLogEntry>()];
+
+    let src = unsafe {
+        let offset = core::mem::size_of::<PcrLogEntry>() * entry_index;
+        let ptr = (&mut PCR_LOG_ORG as *mut u8).add(offset);
+        core::slice::from_raw_parts_mut(ptr, core::mem::size_of::<PcrLogEntry>())
+    };
+
+    pcr_entry.copy_from_slice(src);
+    PcrLogEntry::read_from_prefix(pcr_entry.as_bytes()).unwrap()
+}
+
+fn process_mailbox_commands() {
+    let mbox = caliptra_registers::mbox::RegisterBlock::mbox_csr();
+
+    let cmd = mbox.cmd().read();
+    cprintln!("[fmc] Received command: 0x{:08X}", cmd);
+    match cmd {
+        0x1000_0000 => {
+            read_pcr_log(&mbox);
+        }
+        0x1000_0001 => {
+            // TODO: Generate certs
+        }
+        _ => {
+            // No-op
+        }
+    }
+}
+
+fn read_pcr_log(mbox: &caliptra_registers::mbox::RegisterBlock) {
+    let mut pcr_entry_count = 0;
+    loop {
+        let pcr_entry = get_pcr_entry(pcr_entry_count);
+        if PcrLogEntryId::from(pcr_entry.id) != PcrLogEntryId::Invalid {
```

Comment:
Optional nit: consider inverting this logic and breaking early, can un-indent this whole loop.

```
loop {
  let pcr_entry = get_pcr_entry(pcr_entry_count);
  if PcrLogEntryId::from(pcr_entry.id) == PcrLogEntryId::Invalid {
    break;
  }
  pcr_entry_count += 1;
  ...
}